pkg/server/server_test.go

```diff
@@ -45,6 +45,66 @@ func init() {
 	}
 }
 
+func ExampleNewServerWithOpts() {
+	openfga, err := NewServerWithOpts(WithDatastore(memory.New()))
```

Comment:
The interesting part is here, should we document what options that are typical to be used as library?  Also, do we need to note what sort of middleware we expect user to setup?

Reply:
> should we document what options that are typical to be used as library

I'm not sure if this is the best place to put that, i don't want to be too verbose, or to repeat documentation. We could point to https://github.com/openfga/openfga/blob/c461c42a3993af58fa2b4c60242ef97cb7765d09/.config-schema.json? I'm open to ideas

> need to note what sort of middleware we expect user to setup?

🤔 You don't need middleware to use it as a library. At one point in the past we realized that if you used it as a library you had to add the gRPC validation middleware, but that is not the case anymore.

Reply:
The godoc should serve as the list 😄 . That's the point.

Focus on good godoc and we'll get good documentation on go.pkg.dev

Reply:
Are we good to merge as is then?